ruby/cleanup/action.yml

```
@@ -0,0 +1,39 @@
+name: Dry-Run Cleanup
+description: Cleans up generated tag and release after a dry-run.
+inputs:
+  app_id:
+    description: The APP_ID defined for this project
+    required: true
+  app_private_key:
+    description: The APP_PRIVATE_KEY defined for this project
+    required: true
+  tag:
+    description: The name of the tag (and release) to clean up
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: "Create temporary app token"
+      uses: actions/create-github-app-token@v1
+      id: app-token
+      with:
+        app-id: ${{ inputs.app_id }}
+        private-key: ${{ inputs.app_private_key }}
+
+    - name: "Store GitHub token in environment"
+      shell: bash
+      run: echo "GH_TOKEN=${{ steps.app-token.outputs.token }}" >> "$GITHUB_ENV"
+
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        token: ${{ env.GH_TOKEN }}
```

The `secure-checkout` action will handle these three steps for you:

```suggestion
    - name: "Generate token and check out repository"
      uses: mongodb-labs/drivers-github-tools/secure-checkout@v2
      with:
        app-id: ${{ inputs.app_id }}
        private-key: ${{ inputs.app_private_key }}
```